packages/react-native/scripts/generate.js

```diff
@@ -50,10 +50,26 @@ function generate({ configPath, absolute = false, useJs = false }) {
 
   const doctools = 'require("@storybook/react-native/dist/preview")';
 
-  // TODO: implement presets or something similar
-  const enhancer = main.addons?.includes('@storybook/addon-ondevice-actions')
-    ? "require('@storybook/addon-actions/preview')"
-    : '';
+  const enhancer = [];
+
+  const enhancerDependenciesMap = {
+    '@storybook/addon-ondevice-actions': '@storybook/addon-actions',
+  };
+
+  main.addons?.forEach((_addon) => {
+    const addon = enhancerDependenciesMap[_addon] || _addon;
+    let addonPath;
+    try {
+      addonPath = path.dirname(require.resolve(addon));
+    } catch (error) {
+      console.error(`Failed to resolve addon: ${addon}`, error);
+      return null; // Skip this addon if it cannot be resolved
+    }
+    const isPreviewFileExists = getPreviewExists({ configPath: addonPath });
+    if (isPreviewFileExists) {
+      enhancer.push(`require('${addon}/dist/preview')`);
```

Comment:
Seems like this would be without `dist` in the path otherwise the check would be wrong.

Also we can have `@storybook/addon-ondevice-actions` re-export the actions preview in `@storybook/addon-actions/preview` so that we can follow that pattern.

``` 
enhancer.push(`require('${addon}/preview')`);
```

Reply:
I think if we remove the `/dist` path, the bundler will get the raw code since `node_modules` is ignored. Plus, the source file can be anywhere, but according to the docs, it should be bundled into the dist directory: 
https://storybook.js.org/docs/addons/writing-addons#module-metadata But probably, we could look for a file in the dist directly.

I love the idea of re-exporting `@storybook/addon-actions/preview`. I just wanted to focus on the specific issue I'm facing here and don't touch anything else

Reply:
Well if this is the case then the check should be changed to look for the dist folder

Reply:
I guess you already updated it but i will also check how its handled in storybook core before moving forward

Reply:
> Well if this is the case then the check should be changed to look for the dist folder
> 
> I.e getpreviewpath(addonpath/dist)

It won't work as the package is resolving to the /dist path. Try to play with it a little pls And let me know how to proceed

Reply:
if exports are used like in the docs here
```
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "node": "./dist/index.js",
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    },
    "./manager": "./dist/manager.mjs",
    "./preview": "./dist/preview.mjs",
    "./package.json": "./package.json"
  },
```

then this allows the import from package-name/preview assuming that package exports are enabled for metro which they are required to be for rn storybook 8

Reply:
yep, so now should work as expected